extensions/ql-vscode/src/remote-queries/sarif-processing.ts

```diff
@@ -186,18 +147,149 @@ export function tryGetRule(
   }
 
   return errors;
 }
+
+function validateRegion(region: sarif.Region | undefined): string[] {
+  const errors: string[] = [];
+
+  if (!region) {
+    errors.push('No region found in the SARIF result location');
+    return errors;
+  }
+  if (region.startLine === undefined) {
+    errors.push('No start line set for a result region');
+  }
+  if (region.startColumn === undefined) {
+    errors.push('No start column set for a result region');
+  }
+  if (region.endColumn === undefined) {
+    errors.push('No end column set for a result region');
+  }
+
+  if (errors.length > 0) {
+    return errors;
+  }
+
+  if (region.endLine! === region.startLine! &&
+    region.endColumn! < region.startColumn!) {
+    errors.push('End column is greater than the start column in a result region');
+  }
+
+  return errors;
+}
+
+function extractLocation(
+  location: sarif.Location
+): {
+  processedLocation: Location | undefined,
+  errors: string[]
+} {
+  const message = location.message?.text;
+
+  const errors = [];
+
+  const contextRegion = location.physicalLocation?.contextRegion;
+  const contextRegionErrors = validateContextRegion(contextRegion);
+  errors.push(...contextRegionErrors);
+
+  const region = location.physicalLocation?.region;
+  const regionErrors = validateRegion(region);
+  errors.push(...regionErrors);
+
+  const filePath = location.physicalLocation?.artifactLocation?.uri;
+  if (!filePath) {
+    errors.push('No file path found in the SARIF result location');
+  }
+
+  if (errors.length > 0) {
+    return { processedLocation: undefined, errors };
+  }
+
+  const processedLocation = {
+    message,
+    filePath,
+    codeSnippet: {
+      startLine: contextRegion!.startLine,
+      endLine: contextRegion!.endLine,
+      text: contextRegion!.snippet!.text
+    },
+    highlightedRegion: {
+      startLine: region!.startLine,
+      startColumn: region!.startColumn,
+      endLine: region!.endLine,
+      endColumn: region!.endColumn
+    }
+  } as Location;
+
+  return { processedLocation, errors: [] };
+}
+
+function extractCodeFlows(
+  result: sarif.Result
+): {
+  codeFlows: CodeFlow[],
+  errors: string[]
+} {
+  const codeFlows = [];
+  const errors = [];
+
+  if (result.codeFlows) {
+    for (const codeFlow of result.codeFlows) {
+      const threadFlows = [];
+
+      for (const threadFlow of codeFlow.threadFlows) {
+        for (const location of threadFlow.locations) {
+          const { processedLocation, errors: locationErrors } = extractLocation(location);
+          if (locationErrors.length > 0) {
+            errors.push(...locationErrors);
+            continue;
+          }
+
+          threadFlows.push({
+            filePath: processedLocation!.filePath,
+            codeSnippet: processedLocation!.codeSnippet,
+            highlightedRegion: processedLocation!.highlightedRegion,
+            message: processedLocation!.message
```

Comment:
If my comment above it taken to account, `processedLocation` will never be undefined and the `!` can go away.